src/libertem/analysis/com.py

```diff
@@ -1,17 +1,66 @@
 import logging
-
+import inspect
 import numpy as np
 
 from libertem import masks
 from libertem.viz import CMAP_CIRCULAR_DEFAULT, visualize_simple
 from .base import AnalysisResult, AnalysisResultSet
 from .masks import BaseMasksAnalysis
 from libertem.corrections.coordinates import rotate_deg, flip_y, identity
-
+from .helper import GeneratorHelper
 
 log = logging.getLogger(__name__)
 
 
+class ComTemplate(GeneratorHelper):
+
+    short_name = "com"
+    api = "create_com_analysis"
+    temp = GeneratorHelper.temp_analysis
+    temp_analysis = temp + ["print(com_result)"]
+    channels = [
+        "field",
+        "magnitude",
+        "divergence",
+        "curl",
+        "x",
+        "y"
+    ]
+
+    def __init__(self, params):
+        self.params = params
+
+    def get_docs(self):
+        title = "COM Analysis"
+        from libertem.api import Context
+        docs_rst = inspect.getdoc(Context.create_com_analysis)
+        docs = self.format_docs(title, docs_rst)
+        return docs
+
+    def convert_params(self):
+        params = ['dataset=ds']
+        for k in ['cx', 'cy']:
+            params.append(f'{k}={self.params[k]}')
+        params.append(f"mask_radius={self.params['r']}")
+        return ', '.join(params)
+
+    def get_plot(self):
+        plot = []
+        for channel in self.channels[:3]:
+            plot.append("fig, axes = plt.subplots()")
+            plot.append(f'axes.set_title("{channel}")')
+            plot.append(f'axes.imshow(com_result.{channel}.visualized)')
```

Comment:
This is the last remaining instance that should plot the data instead of the visualized result. :-)

For the field you can use the Cubehelix plot from èmpyre` like for Radial Fourier Analysis, and the two other ones are just `imshow()`.